### PR TITLE
Fix for Rails 4.2 and type_cast

### PR DESCRIPTION
**Pull request by pschambacher**
@grosser 

Code was trying to do something smart to avoid limitations of type_cast_for_database which rejects values larger than the column can handle. This does not work with datetime column so going for something simpler and changing the test rather than the code.

### Risks
 - None

### CODE REVIEW

File: test/configuration_mixin_test.rb [hunk 1]
```diff
@@ -18,10 +18,8 @@
     end
 
     it "not generate keys longer that 255" do
-      very_large_number = (1..999).to_a.join
-      query_attributes  = [ [:blog_id, very_large_number] ]
-
-      assert(Post.kasket_key_for(query_attributes).size < 255)
+      Post.stubs(:quoted_value_for_column).returns((1..999).to_a.join.to_s)
+      assert(Post.kasket_key_for([:blog_id, 1]).size < 255)
```
Comment (grosser) [inline test/configuration_mixin_test.rb]: idk ... silly test/feature anyway since memcached will auto-shorten long keys ...

Comment (grosser) [inline test/configuration_mixin_test.rb]: hmm I guess the prefix is nice ...

Comment (grosser) [inline test/configuration_mixin_test.rb]: nicer errors with `xxx.must_be :<, 255`

**Comment by grosser** [comment]
:+1:

File: test/configuration_mixin_test.rb (hunk 1 above)
Comment (staugaard) [inline test/configuration_mixin_test.rb]: memcached will not auto-shorten keys. Some memcached clients do, but not all.

Comment (pschambacher) [inline test/configuration_mixin_test.rb]: I'll need to revert the 4.0.0 commit and tag. This should be included first.

Comment (grosser) [inline test/configuration_mixin_test.rb]: might as well do 4.0.1 ... 4.0.0 was not released anyway

**Comment by pschambacher** [comment]
Still can't release kasket on rubygem :sob:

**Comment by staugaard** [comment]
@pschambacher shoot me your rubygems.org email address, and I'll add you.

**Comment by pschambacher** [comment]
Should be my Zendesk email

**Comment by staugaard** [comment]
added

**Comment by pschambacher** [comment]
Worked. Thanks @staugaard

**Pull request merged**